### PR TITLE
feat: secp256k1 fromSeedPhrase now supported

### PR DESCRIPTION
**Pull request by krpeacock**
# Description

For some time, people have requested support for re-using a consistent identity across their stack, either for testing purposes or CI. This code [has been available](https://forum.dfinity.org/t/using-dfinity-agent-in-node-js/6169/41?u=kpeacock), but was not included in `agent-js`. For convenience, this method is now supported

Affects #622 

# How Has This Been Tested?

Unit testing

# Checklist:

- [ ] My changes follow the guidelines in [CONTRIBUTING.md](https://github.com/dfinity/agent-js/blob/main/CONTRIBUTING.md).
- [ ] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I have edited the CHANGELOG accordingly.
- [ ] I have made corresponding changes to the documentation.

### CODE REVIEW

**Review by ericswanson-dfinity — APPROVED** [review]

**Comment by krpeacock** [comment]
Pending security review

**Review by robin-kunzler — COMMENTED** [review]
@krpeacock : Thanks a lot for adding this! As requested I reviewed the security warnings and they look good to me, thanks. Also, given those warnings it seems OK to me to provide this additional features. 

Just out of curiosity: Do you know what use cases people want to solve with this? 

FYI Internet Identity also has code that creates a Ed25519KeyIdentity from a seed phrase. See [here](https://github.com/dfinity/internet-identity/blob/df8e52a97055bcd767439e624703d6d26758cf9f/src/frontend/src/crypto/ed25519.ts#L14).

File: packages/identity/src/identity/secp256k1.ts [hunk 1]
```diff
@@ -132,6 +134,33 @@ export class Secp256k1KeyIdentity extends SignIdentity {
...
+    // Warn if provided phrase is not conventional
+    if (phrase.split(' ').length < 12 || phrase.split(' ').length > 24) {
+      console.warn(
+        'Warning - an unusually formatted seed phrase has been provided. Decoding may not work as expected',
+      );
+    }
+
+    const seed = mnemonicToSeedSync(phrase, password);
+    const root = hdkey.fromMasterSeed(seed);
+    root;
```
Comment (krpeacock) [inline packages/identity/src/identity/secp256k1.ts:158]:
```suggestion
```

[the remark this answers is not in the document]
Comment (krpeacock) [reply packages/identity/src/identity/secp256k1.ts]: yeah, that's a leftover side-effect-less log. I need to find a way to lint for those

**Pull request merged**